lib/TopStoriesFeed.jsm

```diff
@@ -578,6 +610,18 @@ this.TopStoriesFeed = class TopStoriesFeed {
         if (action.data.name === "pocketCta") {
           this.dispatchPocketCta(action.data.value, true);
         }
+        if (action.data.name === OPTIONS_PREF) {
+          try {
+            const options = JSON.parse(action.data.value);
+            if (this.processAffinityProividerVersion(options)) {
+              await this.clearCache();
+              this.uninit();
```

Comment:
What's the purpose of uninit then re-init upon the version pref change? If it's merely for resetting some states, this sounds a bit overkill to me. Another question is do we really need to support two versions of personalization at the same time? Even if the answer is yes, that could be done separately instead of complicating this experiment.

Reply:
We do need this 3 way comparison. Off is our baseline for organic recommendations, but v1 is the baseline for sponsored content. This allows us to check the behavior across both. If we reach parity with v1 we plan to kill it completely in Q1.

Reply:
Shield doesn't change prefs right away, so if this pref changes, it means there is already a moment where v1 or a non personalized version has started to be generated. It is my understanding (based on other areas of the code) that the only way to regenerate everything with new pref is an uninit/init.